dvc/stage.py

```diff
@@ -794,6 +794,8 @@ def save(self):
 
         self.md5 = self._compute_md5()
 
+        stage_cache.save(self)
```

Comment:
Maybe, we could make it explicit here that it's only run for pipeline stages?
```suggestion
        if not self.is_data_source:
             stage_cache.save(self)
```

Reply:
Also need to account for imports and others. But yeah, good point. In stage cache itself, we ignore caching if there is no `cmd`, so we should do the same here.